### PR TITLE
Added support for `webkitgtk2-4.0` and added `GI_TYPELIB_PATH` for gi path

### PR DESCRIPTION
**Pull request by soumyaDghosh**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [x] Have you successfully run `pytest tests/unit`?

-----

All the tests are mostly passing, except one fail in the test. Probably not my fault

```
_______________________________________________________________________ test_get_extensions_data_dir ________________________________________________________________________

    def test_get_extensions_data_dir():
>       assert (get_extensions_data_dir() / "desktop").is_dir()
E       AssertionError: assert False
E        +  where False = <bound method Path.is_dir of PosixPath('/home/ubuntu/.venv/snapcraft/share/snapcraft/extensions/desktop')>()
E        +    where <bound method Path.is_dir of PosixPath('/home/ubuntu/.venv/snapcraft/share/snapcraft/extensions/desktop')> = (PosixPath('/home/ubuntu/.venv/snapcraft/share/snapcraft/extensions') / 'desktop').is_dir
E        +      where PosixPath('/home/ubuntu/.venv/snapcraft/share/snapcraft/extensions') = get_extensions_data_dir()

tests/unit/extensions/test_extensions.py:280: AssertionError
```

### CODE REVIEW

File: snapcraft/extensions/gnome.py [hunk 1]
```diff
@@ -155,8 +155,11 @@ def get_root_snippet(self) -> Dict[str, Any]:
                 }
             },
             "layout": {
-                "/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0": {
-                    "bind": "$SNAP/gnome-platform/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0"
+                "/usr/lib/$CRAFT_ARCH_TRIPLET/webkit2gtk-4.0": {
+                    "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET/webkit2gtk-4.0"
+                },
+                "/usr/lib/$CRAFT_ARCH_TRIPLET/webkit2gtk-4.1": {
+                    "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET/webkit2gtk-4.1"
```
Comment (lengau) [inline snapcraft/extensions/gnome.py:162]:
```suggestion
                "/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.0": {
                    "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.0"
                },
                "/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.1": {
                    "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.1"
```

`CRAFT_ARCH_TRIPLET` will be deprecated in snapcraft 8 :-)

File: snapcraft/extensions/gnome.py [hunk 2]
```diff
@@ -240,6 +243,15 @@ def get_part_snippet(self, *, plugin_name: str) -> Dict[str, Any]:
                         ],
                     ),
                 },
+                {
+                    "GI_TYPELIB_PATH": prepend_to_env(
+                        "GI_TYPELIB_PATH",
+                        [
+                            f"/snap/{sdk_snap}/current/usr/lib/girepository-1.0",
+                            f"/snap/{sdk_snap}/usr/lib/$CRAFT_ARCH_TRIPLET/girepository-1.0",
```
Comment (lengau) [inline snapcraft/extensions/gnome.py:251]:
```suggestion
                            f"/snap/{sdk_snap}/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/girepository-1.0",
```

File: tests/unit/extensions/test_gnome.py [hunk 3]
```diff
@@ -93,8 +93,11 @@ def test_get_root_snippet(gnome_extension):
             }
         },
         "layout": {
-            "/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0": {
-                "bind": "$SNAP/gnome-platform/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0"
+            "/usr/lib/$CRAFT_ARCH_TRIPLET/webkit2gtk-4.0": {
+                "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET/webkit2gtk-4.0"
+            },
+            "/usr/lib/$CRAFT_ARCH_TRIPLET/webkit2gtk-4.1": {
+                "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET/webkit2gtk-4.1"
```
Comment (lengau) [inline tests/unit/extensions/test_gnome.py:100]:
```suggestion
                "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.0"
            },
            "/usr/lib/$CRAFT_ARCH_TRIPLET/webkit2gtk-4.1": {
                "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.1"
```

File: tests/unit/extensions/test_gnome.py [hunk 4]
```diff
@@ -141,8 +144,11 @@ def test_get_root_snippet_with_external_sdk(gnome_extension_with_build_snap):
...
```
Comment (lengau) [inline tests/unit/extensions/test_gnome.py:151]:
```suggestion
            "/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.0": {
                "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.0"
            },
            "/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.1": {
                "bind": "$SNAP/gnome-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/webkit2gtk-4.1"
```

File: tests/unit/extensions/test_gnome.py [hunk 5]
```diff
@@ -249,6 +255,15 @@ def assert_get_part_snippet(gnome_instance):
                     )
                     + "${PYTHONPATH:+:$PYTHONPATH}"
                 },
+                {
+                    "GI_TYPELIB_PATH": ":".join(
+                        [
+                            "/snap/gnome-42-2204-sdk/current/usr/lib/girepository-1.0",
+                            "/snap/gnome-42-2204-sdk/usr/lib/$CRAFT_ARCH_TRIPLET/girepository-1.0",
```
Comment (lengau) [inline tests/unit/extensions/test_gnome.py:262]:
```suggestion
                            "/snap/gnome-42-2204-sdk/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/girepository-1.0",
```

File: tests/unit/extensions/test_gnome.py [hunk 6]
```diff
@@ -312,6 +327,15 @@ def test_get_part_snippet_with_external_sdk(gnome_extension_with_build_snap):
                 )
                 + "${PYTHONPATH:+:$PYTHONPATH}"
             },
+            {
+                "GI_TYPELIB_PATH": ":".join(
+                    [
+                        "/snap/gnome-44-2204-sdk/current/usr/lib/girepository-1.0",
+                        "/snap/gnome-44-2204-sdk/usr/lib/$CRAFT_ARCH_TRIPLET/girepository-1.0",
```
Comment (lengau) [inline tests/unit/extensions/test_gnome.py:334]:
```suggestion
                        "/snap/gnome-44-2204-sdk/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/girepository-1.0",
```

**Review by lengau — CHANGES_REQUESTED** [review]

**Review by mr-cal — CHANGES_REQUESTED** [review]
Thanks @soumyaDghosh!

It looks like the unit tests for `tests/unit/extensions/test_gnome.py::test_get_root_snippet` are still failing.  Do you mind taking a look?

Also, I see a linter failure for lines that are too long in `snapcraft/extensions/gnome.py` and in the unit tests.  Do you mind breaking those strings into multiple lines to satisfy the linter?

**Comment by soumyaDghosh** [comment]
@mr-cal The tests should pass now. But regarding the lint, complaining about long lines, I don't think I can help it. Those lines it's saying to break (eg: [159](https://github.com/soumyaDghosh/snapcraft/blob/main/snapcraft/extensions/gnome.py#L159)) should be as they are as much I can think. Kindly give your opinion on it.

**Comment by mr-cal** [comment]
> @mr-cal The tests should pass now. But regarding the lint, complaining about long lines, I don't think I can help it. Those lines it's saying to break (eg: [159](https://github.com/soumyaDghosh/snapcraft/blob/main/snapcraft/extensions/gnome.py#L159)) should be as they are as much I can think. Kindly give your opinion on it.

The tests look good, thanks.

I think you can satisfy the linter by breaking those strings into multiple lines.  You can cherry-pick this commit if you'd like: https://github.com/snapcore/snapcraft/commit/ab97ec7b8d96b31a8dde6cd5256bd56e180c8641

Otherwise, I'm happy to push that commit and update this branch onto the latest changes on `main`.

**Comment by soumyaDghosh** [comment]
> 
> The tests look good, thanks.
> 
> I think you can satisfy the linter by breaking those strings into multiple lines.  You can cherry-pick this commit if you'd like: https://github.com/snapcore/snapcraft/commit/ab97ec7b8d96b31a8dde6cd5256bd56e180c8641
> 
> Otherwise, I'm happy to push that commit and update this branch onto the latest changes on `main`.

Sure please push the commit and update this branch. I didn't know it's possible. Thank you.

**Review by lengau — APPROVED** [review]
Looks good, thanks!

**Comment by sergiusens** [comment]
@mr-cal please watch your merge messaging

**Comment by mr-cal** [comment]
Whoops, sorry!

**Pull request merged**